### PR TITLE
Edit cells inline or popup 'intelligent' switch #added

### LINKED ISSUE #508: Have a preference to "Force LONGTEXT to be edited inline, hold option while double clicking for popup view"
I, honestly, have wanted this since this app was Sequel Pro. I have **so many** (almost all) fields (especially in Wordpress) that are labeled as `LONGTEXT` when they're honestly just a few bytes long (sometimes even a boolean). You know what would be even smarter, for Sequel Pro to **detect** the amount of bytes a field is actually using and edit it inline if it's within a number of characters (set by the user but default could be something sane). What do you guys think?

### PR DESCRIPTION
**Pull request by jamesstout**
## Changes:
- New pref to switch between editing 'text' inline to editing in the popup, if the text is over a certain length, also set in prefs. Defaults to 15.

## Closes following issues:
- Closes #508

## Tested:
- Processors:
  - [x] Intel
  - [ ] Apple Silicon
- macOS Versions:
  - [ ] 10.12.x (Sierra)
  - [ ] 10.13.x (High Sierra)
  - [ ] 10.14.x (Mojave)
  - [ ] 10.15.x (Catalina)
  - [x] 11.x (Big Sur)
- Xcode Version: 12.4 (12D4e)
- Localizations:
  - [ ] English
  - [ ] Spanish
  - [ ] Other (please specify)
## Screenshots:

Screenshot 2021-02-25 at 9 16 21 PM

### CODE REVIEW

File: Source/Views/TableViews/SPCopyTable.m [hunk 1]
```diff
@@ -1234,13 +1234,19 @@ - (BOOL)shouldUseFieldEditorForRow:(NSUInteger)rowIndex column:(NSUInteger)colIn
 {
 	// Return YES if the multiple line editing button is enabled - triggers sheet editing on all cells.
 	if ([prefs boolForKey:SPEditInSheetEnabled]) return YES;
-	
+
+    NSUInteger editInSheetForLongTextLengthThreshold = NSUIntegerMax;
```
Comment (Kaspik) [inline Source/Views/TableViews/SPCopyTable.m:1238]: I would have like 15 here (the default), and add the value from prefs to the if condition too and assign it only if it exists - current implementation could lead to getting undefined value from prefs that could cause a crash

Comment (jamesstout) [reply Source/Views/TableViews/SPCopyTable.m:1238]: done

File: Source/Views/TableViews/SPCopyTable.m [hunk 2]
```diff
@@ -1234,13 +1234,25 @@ - (BOOL)shouldUseFieldEditorForRow:(NSUInteger)rowIndex column:(NSUInteger)colIn
 {
 	// Return YES if the multiple line editing button is enabled - triggers sheet editing on all cells.
 	if ([prefs boolForKey:SPEditInSheetEnabled]) return YES;
-	
+
+    NSMutableDictionary *preferenceDefaults = [NSMutableDictionary dictionaryWithContentsOfFile:[[NSBundle mainBundle] pathForResource:SPPreferenceDefaultsFile ofType:@"plist"]];
+
+    NSUInteger editInSheetForLongTextLengthThreshold = (NSUInteger)[preferenceDefaults safeObjectForKey:@"ssfsfs"];
+
+    if([prefs boolForKey:SPEditInSheetForLongText] && [prefs objectForKey:SPEditInSheetForLongTextLengthThreshold]){
```
Comment (Kaspik) [inline Source/Views/TableViews/SPCopyTable.m:1242]: This still doesn't check if `[prefs objectForKey:SPEditInSheetForLongTextLengthThreshold]` is `integerValue` as you are getting on next line... should we do the check?

**Review by Kaspik — APPROVED** [review]

**Comment by olimortimer** [comment]
Thank you.  So if ticked and set to 15, does that mean at 15 it will switch to inline, or switch to pop-up?  My setting was unticked, yet everything was loading in a pop-up, even when choosing to edit inline in the table view.

**Comment by Jason-Morcos** [comment]
> Thank you. So if ticked and set to 15, does that mean at 15 it will switch to inline, or switch to pop-up? My setting was unticked, yet everything was loading in a pop-up, even when choosing to edit inline in the table view.

I _think_ the checkbox is like "always edit in popup" and the length is the length to automatically switch from popup to inline editing regardless of the checkbox. The UI is indeed a little confusing

**Comment by olimortimer** [comment]
Thank you.  Unfortunately I've played around with the settings, and it still doesn't seem to work correctly - closest I've got is double clicking and it edits in a pop-up, click save then it starts editing inline, but then when I edit it pops-up again to edit.

**Pull request merged**